### ISSUE TITLE
injection scope selector of type meta.object-literal not working

### ISSUE THREAD

**Issue by sheetalkamat** [Microsoft/vscode-textmate#28]
I was trying to update and see if [natewallace/language-vscode-javascript-angular2](https://github.com/natewallace/language-vscode-javascript-angular2/) extension works with the new typescript grammar by changing [template.ng.json](https://github.com/natewallace/language-vscode-javascript-angular2/blob/master/syntaxes/template.ng.json#L3) to use `meta.object-literal` instead of `meta.block` injection scope selector. The extension doesn't work because I think it treats the dash as minus operator. I modified the grammar and the injection rule from `meta.object-literal` to `meta.objectliteral` and the extension works as expected.

Note this is the typescript code I tried:

``` typescript
import {Component, HostListener, Directive} from 'angular2/core';

@Directive({selector: 'button[counting]'})
class CountClicks {
  numberOfClicks = 0;
  @HostListener('click', ['$event.target']) onClick(btn) {
    console.log("button", btn, "number of clicks:", this.numberOfClicks++);
  }
}
@Component({
  selector: 'my-app',
  template: `<button counting>Increment</button>`,
  directives: [CountClicks]
})
export class AppComponent {
  constructor() { console.clear(); }
}
```

With the extension working the line should have colors from the scope of tags

```
template: `<button counting>Increment</button>` 
```

**Comment by mjbvz** [comment]
Fixed by ecade6c4385e69199b2b81ccbf27f3c543434c17 I believe

**Comment by alexandrudima** [comment]
@mjbvz :+1: Is ecade6c4385e69199b2b81ccbf27f3c543434c17 something you'd consider cherry picking to master ?

**Comment by mjbvz** [comment]
Yep, I published a minor version update with the fix and have already pulled it into vscode for 1.23

**Comment by alexandrudima** [comment]
@mjbvz ... but ecade6c is not in master on this repository

![image](https://user-images.githubusercontent.com/5047891/38523853-0b413a4c-3c4d-11e8-8615-da96f0ce4144.png)

I see the tags on this repo are also not respecting the npm published version anymore for v3.3.2

Can we please do the following in the future when publishing to npm:

1. `npm run prepublish`
2. `npm version minor` or `npm version patch` as needed. this will modify `package.json`
3. `git push && git push --tags`
4. `npm publish`.

Following these simple steps will help that others can easily correlate npm releases with tags and with commits and that the changes are publicly visible.

**Comment by mjbvz** [comment]
Whoops, pushed to my origin instead of this one. Just pushed here too.

Sorry for the confusion. Will make sure we also have the correct tag
